doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,487 @@
+
+.. code-block:: c
+
+    // Constructing a single record from separate values
+    ddlog_record **struct_args;
+    struct_args = (ddlog_record**)malloc(3 * sizeof(ddlog_record*));
+    struct_args[0] = src;
+    struct_args[1] = dst;
+    struct_args[2] = lstatus;
+    ddlog_record *new_record = ddlog_struct("Links", struct_args, 3);
+
+Once, we have DDlog values in place we need to construct a record that we will insert into the :code:`Links` relation.
+For that purpose we can use :code:`ddlog_struct()` function.
+It takes three arguments:
+
+#. Name of the relation, which this record belongs to.
+#. A pointer to an array of :code:`ddlog_record`'s.
+#. Length of the aformentioned array.
+
+You don't have to dynamically allocate memory for the :code:`struct_args` in this example.
+As we know exactly how many elements this array contains we can simply put it on the stack.
+
+.. code-block:: c
+
+    // Let's print the record that we are about to insert
+    // to the `Links` relation
+    char *record_to_insert_as_string = ddlog_dump_record(new_record);
+    printf("Inserting the following record: %s\n", record_to_insert_as_string);
+    ddlog_string_free(record_to_insert_as_string);
+
+Using the :code:`ddlog_dump_record()` function from DDlog API we can retrieve the record as a string and print it to make sure everything looks right.
+Note, that this function will allocate some memory on a heap and we are responsible for freeing it.
+
+.. code-block:: c
+
+    // Start transaction
+    if (ddlog_transaction_start(prog) < 0) {
+        fprintf(stderr, "failed to start transaction\n");
+        exit(EXIT_FAILURE);
+    };
+
+As mentioned before, DDlog is a transaction-based programming language.
+Thus, before inserting a new record into a relation we need to start transaction.
+
+.. code-block:: c
+
+    // Create `insert` command
+    ddlog_cmd *cmd = ddlog_insert_cmd(LinksTableID, new_record);
+    if (cmd == NULL) {
+        fprintf(stderr, "failed to create insert command\n");
+        exit(EXIT_FAILURE);
+    }
+
+Next step is to create an insertion DDlog command using :code:`ddlog_insert_cmd()` function.
+We need to provide this function with the table ID of a target relation and a record that we want to insert into that relation.
+
+.. code-block:: c
+
+    // Apply updates to the relation with records
+    // specified in the provided command `cmd`
+    if (ddlog_apply_updates(prog, &cmd, 1) < 0) {
+        fprintf(stderr, "failed to apply updates\n");
+        exit(EXIT_FAILURE);
+    };
+
+Once we have command ready we can apply it to our DDlog program using :code:`ddlog_apply_updates()` function.
+We need to supply program handle, array of commands to be applied, and the length of this array.
+In our case, we only have a single command to apply.
+However, it is possible to pass multiple commands to a single call to the :code:`ddlog_apply_update()` function.
+It helps avoiding cluttering code if you want to apply multiple commands.

```

Comment:
I would actually say that often applying multiple commands in a single transaction can be more efficient (batching).